Tests dont compile

Compiling libint with testing on results in a crash
```
/builddir/build/BUILD/libint2-2.13.1-build/libint-2.13.1-post999/tests/unit/test-core-ints.cc:72:41:   required from here
   72 |               m, sstream_convert<double>(Tref),
      |                  ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
/builddir/build/BUILD/libint2-2.13.1-build/libint-2.13.1-post999/include/libint2/numeric.h:208:23: error: cannot convert ‘const char*’ to ‘double’ in initialization
  208 |   To to(ss.str().c_str());
      |         ~~~~~~~~~~~~~~^~
      |                       |
      |                       const char*
```

--- a/include/libint2/numeric.h
+++ b/include/libint2/numeric.h
@@ -198,6 +198,22 @@ inline int get_max_digits10(const Real& value) {
   return std::numeric_limits<Real>::max_digits10;
 }
 
+namespace detail {
+template <typename To>
+typename std::enable_if<std::is_constructible<To, const char*>::value, To>::type
+sstream_convert_extract(std::stringstream& ss) {
+  return To(ss.str().c_str());
+}
+
+template <typename To>
+typename std::enable_if<!std::is_constructible<To, const char*>::value, To>::type
+sstream_convert_extract(std::stringstream& ss) {
+  To to;
+  ss >> to;
+  return to;
+}
+}  // namespace detail
+
 template <typename To, typename From>
 typename std::enable_if<!std::is_same<typename std::decay<To>::type,
                                       typename std::decay<From>::type>::value,
@@ -205,8 +221,7 @@ typename std::enable_if<!std::is_same<typename std::decay<To>::type,
 sstream_convert(From&& from) {
   std::stringstream ss;
   ss << std::scientific << std::setprecision(get_max_digits10(from)) << from;
-  To to(ss.str().c_str());
-  return to;
+  return detail::sstream_convert_extract<To>(ss);
 }
 
 template <typename To>

